src/Bishop.cpp

```diff
@@ -1,6 +1,40 @@
-#include "Bishop.h"
-
-Bishop::Bishop(const string& _ch)
-{
-
-}
+#include "Bishop.h"
+#include "FactoryPieces.h"
+#include "Board.h"
+
+bool Bishop::m_registerit = FactoryPieces::registerit(BISHOP_CHAR, [](const char& _ch, const Location& loc, PlayerColor color, Board* board)

```

Comment:
These register calls that happen for every piece are A LOT of code duplication.
Maybe you can reduce it somehow? Does every piece have to register itself into the factory? Why not just have a map between chars and their respective classes? Then you could just call the Class constructor when needed